File: update-diamondUtils.js
const fs = require('fs');
const file = 'artifacts/iic-study-app-replit/src/utils/diamondUtils.ts';
let content = fs.readFileSync(file, 'utf8');

const target = `export const DIAMOND_SUBSCRIPTION_PLANS: DiamondSubscriptionPlan[] = [`;
const replacement = `export const PRESET_DIAMOND_SUB_TEMPLATES = [
  {
    id: '10_DIA_BASE',
    name: 'Starter Diamond Pass',
    dailyDiamonds: 10,
    ratePerDiamond: 2,
    badge: '10 💎 / DAY',
    description: 'Basic unlock pass (10 Diamonds / Day)',
  },
  {
    id: '25_DIA_BASE',
    name: 'Popular Diamond Pass',
    dailyDiamonds: 25,
    ratePerDiamond: 2,
    badge: '25 💎 / DAY',
    description: 'Perfect for regular users (25 Diamonds / Day)',
  },
  {
    id: '50_DIA_BASE',
    name: 'Pro Diamond Pass',
    dailyDiamonds: 50,
    ratePerDiamond: 2,
    badge: '50 💎 / DAY',
    description: 'For power users (50 Diamonds / Day)',
  },
  {
    id: '100_DIA_BASE',
    name: 'Ultra Mega Pass',
    dailyDiamonds: 100,
    ratePerDiamond: 2,
    badge: '100 💎 / DAY',
    description: 'Maximum unlocks (100 Diamonds / Day)',
  },
];

export const DIAMOND_SUBSCRIPTION_PLANS: DiamondSubscriptionPlan[] = [`;

content = content.replace(target, replacement);
fs.writeFileSync(file, content);

File: update_diamond_utils.py
import re

with open('artifacts/iic-study-app-replit/src/utils/diamondUtils.ts', 'r') as f:
    content = f.read()

new_func = """export function getDiamondSubscriptionPlans(settings?: SystemSettings): DiamondSubscriptionPlan[] {
  if (settings?.diamondSubscriptionPlans && settings.diamondSubscriptionPlans.length > 0) {
    return settings.diamondSubscriptionPlans;
  }
  return DIAMOND_SUBSCRIPTION_PLANS;
}

export function activateDiamondSub(
  user: User,
  planId: string,
  customDurationDays?: number,
  customPlanName?: string,
  settings?: SystemSettings
): User {
  const plans = getDiamondSubscriptionPlans(settings);
  const plan = plans.find(p => p.id === planId) || plans[0];"""

# Replace activateDiamondSub signature and add getDiamondSubscriptionPlans
find_str = """export function activateDiamondSub(
  user: User,
  planId: string,
  customDurationDays?: number,
  customPlanName?: string
): User {
  const plan = DIAMOND_SUBSCRIPTION_PLANS.find(p => p.id === planId) || DIAMOND_SUBSCRIPTION_PLANS[0];"""

content = content.replace(find_str, new_func)

# Fix imports in diamondUtils
import_find = "import { User, UserDiamondSubscription } from '../types';"
import_replace = "import { User, UserDiamondSubscription, SystemSettings } from '../types';"
content = content.replace(import_find, import_replace)

with open('artifacts/iic-study-app-replit/src/utils/diamondUtils.ts', 'w') as f:
    f.write(content)

print("Updated diamondUtils.ts")
